crates/fluvio-benchmark/src/benchmark_config/benchmark_matrix.rs

```diff
@@ -0,0 +1,189 @@
+                b.producer_batch_size = Some(v);
+            })
+            .cross_iterate(&self.producer_queue_size, |v, b| {
+                b.producer_queue_size = Some(v);
+            })
+            .cross_iterate(&self.producer_linger_millis, |v, b| {
+                b.producer_linger = Some(Duration::from_millis(v));
+            })
+            .cross_iterate(&self.producer_server_timeout_millis, |v, b| {
+                b.producer_server_timeout = Some(Duration::from_millis(v));
+            })
+            .cross_iterate(&self.producer_compression, |v, b| {
+                b.producer_compression = Some(v);
+            })
+            .cross_iterate(&self.consumer_max_bytes, |v, b| {
+                b.consumer_max_bytes = Some(v);
+            })
+            .cross_iterate(&self.num_concurrent_producer_workers, |v, b| {
+                b.num_concurrent_producer_workers = Some(v);
+            })
+            .cross_iterate(&self.num_concurrent_consumers_per_partition, |v, b| {
+                b.num_concurrent_consumers_per_partition = Some(v);
+            })
+            .cross_iterate(&self.num_partitions, |v, b| {
+                b.num_partitions = Some(v);
+            })
+            .cross_iterate(&self.record_size, |v, b| {
+                b.record_size_strategy = Some(v);
+            })
+            .cross_iterate(&self.record_key_allocation_strategy, |v, b| {
+                b.record_key_allocation_strategy = Some(v);
+            })
+            .build()
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, Copy, Clone, Hash, PartialEq, Eq)]
+pub enum RecordKeyAllocationStrategy {
+    /// RecordKey::NULL
+    NoKey,
+    /// All producer workers will use the same key
+    AllShareSameKey,
+
+    /// Each producer will use the same key for each of their records
+    ProducerWorkerUniqueKey,
+
+    /// Each producer will round robin from 0..N for each record produced
+    RoundRobinKey(u64),
+
+    RandomKey,
+}
+
+pub fn get_default_config() -> Vec<BenchmarkMatrix> {
+    walkdir::WalkDir::new(DEFAULT_CONFIG_DIR)
+        .into_iter()
+        .filter_map(|e| e.ok())
+        .filter_map(|e| {
+            if e.path().extension().is_some() {
+                Some(e)
+            } else {
+                None
+            }
+        })
+        .map(|e| {
+            let file = File::open(e.path()).unwrap();
+            serde_yaml::from_reader::<_, BenchmarkMatrix>(file).unwrap()
+        })
+        .collect()
+}
+
+pub fn get_config_from_file(path: &str) -> Vec<BenchmarkMatrix> {
+    let file = File::open(path).unwrap();
+    vec![serde_yaml::from_reader::<_, BenchmarkMatrix>(file).unwrap()]
+}
+
+#[cfg(test)]
+mod tests {
+
+    use std::{path::Path, fs::File};
+
+    use super::BenchmarkMatrix;
+
+    fn test_config(path: &Path) -> bool {
+        let file = File::open(path).expect("Unable to open file");
+        match serde_yaml::from_reader::<_, BenchmarkMatrix>(file) {
```

Comment:
replace with assert

Reply:
Yes, I was thinking about this a little as I know you prefer non-iterative tests. 

In this case, it needs to be dynamic because it is testing the config files in the default directory that will be run when the command is executed without arguments. The idea is to prevent shipping invalid config files to the user. 

The reason this is not an assert is that it can test all of the default config files at once and report on the status of each file independent of each other, including a helpful message for what needs to be fixed and where. Otherwise it will only test one and fail.

It can be changed still to an assert if you want, but then if multiple files are need updating after a change to the config struct you have to re-run the tests over and over again to catch all of them.

Reply:
Default configs are now hard-coded for packaging, so no longer relevant